docs/20_smart-contracts/40_guides/10_create_smart_contracts_guide.md

```diff
@@ -0,0 +1,1351 @@
+
+CONTRACT hello : public contract {
+   public:
+      using contract::contract;
+
+      ACTION hi( name nm );
+};
+```
+
+A smart contract class definition must:
+
+- Be annotated by the [[eosio::contract]] attribute which tells the compiler it is a smart contract class; in the `hello.hpp` generated code the `CONTRACT` macro is used, which expands to `class [[eosio::contract]]` C++ code.
+- Be derived from the `contract` class which provides basic smart contract functionality.
+- Define at least a public action function, in this case it has only one, the `hi` action.
+- Define a public action wrapper for each action defined.
+
+You will learn more about actions later.
+
+### The hello.cpp File
+
+The `hello.cpp` is the C++ implementation file and it contains the `hello` smart contract C++ class implementation, that is, the implementation for every member function of the class.
+
+## Actions
+
+An action is a method, defined and implemented by a smart contract class, that can be called by an authorized account to perform some specific task or operation on the blockchain. Actions can be called by other contracts or by external accounts with the EOS Chain API.
+
+Actions can have parameters and return values and their responsibility is to execute the business logic of the contract. For example, an action in a token contract might transfer tokens from one account to another or update the balances of multiple accounts.
+
+Each action requires a specific level of authorization, which can be specified in the action's code. This ensures that only authorized accounts can execute specific actions.
+
+The `hello` smart contract class has only one action implemented by its `hi` public member function.
+
+```c++
+#include <hello.hpp>
+
+ACTION hello::hi( name nm ) {
+   /* fill in action body */
+   print_f("Name : %\n",nm);
+}
+```
+
+The function that implements a smart contract action must be annotated by the `[[eosio::action("action.name")]]` attribute. The `action.name` is optional and if not specified then the action is named by the function name that implements it.
+In the `hello.cpp` generated code the `ACTION` macro is used which expands to `[[eosio::action]] void` C++ code.
+
+The name of the action must:
+
+- Be no longer than 13 characters.
+- Contain only `.`, `a`-`z`, or `1`-`5` characters.
+- May not end with `.`.
+
+Note that when you use the `ACTION` macro the action name is the same as the function name that implements it. Because of that the action name inherits the limitations of the C++ function names as well, which means it can not have `.` in it.
+If you use the `[[eosio::action("action.name")]]` attribute you can name the action differently than the function name that implements it.
+
+## Inline Actions
+
+An inline action is an action that is executed within the context of a parent action. This means that an inline action is initiated by a smart contract action and is executed within the same transaction as the parent action.
```

Comment:
Here's another example of simplifying things to make it shorter, the next 3 paragraphs could just be: 

"An inline action is an action called from within a smart contract.

Inline actions make sure that a second action is always executed within a transaction when the first action is called. 

If any part of the transaction fails, the inline action will unwind with the rest of the transaction."

I also removed the part about order since inline ordering is complicated and should be in the advanced smart contracts guide.

Reply:
fixed